examples/EncodeExample.res

```diff
@@ -42,8 +42,8 @@ module Encode = {
 
   let line = (~points, ~thickness=?, ()) =>
     Unsafe.object({
-      "points": array(point, points),
-      "thickness": option(int, thickness),
+      "points": array(point)(points),
+      "thickness": option(int)(thickness),
```

Comment:
I'm hesitant to change the examples, since this gives the wrong impression of what the API is meant to be. I wonder if it would be better to just make another major release in order to make these uncurried right away instead.

Reply:
If it's not built on npm install, there's no hurry whatsoever for this.

Reply:
Removed this change.

A side comment here. One thing I also tested, is what happens when one defines:
```res
@unboxed
type t<'a> = {encode: 'a => Js.Json.t}

```
And this would look like:
```res
array(point).encode(points)
```

Reply:
Is there a practical benefit to this? The reason I haven't just made it abstract, like in `Decode`, is that it wouldn't be possible to expose the primitives as `%identity` externals then, meaning it would incur significantly more overhead. And this would have the same effect.